Data Explorer: Generate download script (#10)
seqeralabs/docs: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx

You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
:::

### Download multiple files

You can download multiple files via the user interface or using a script in your cloud provider's CLI. 

> Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**. The code snippet provided is specific to the cloud provider you've configured. You may be prompted to authenticate during the download process.
> 
> Refer to your cloud provider's documentation for troubleshooting credential-related issues:

Reviewer: ```suggestion
Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**. 

The code snippet provided is specific to the cloud provider you've configured. You may be prompted to authenticate during the download process. Refer to your cloud provider's documentation for troubleshooting credential-related issues:
```
Mainly because it seems likely that we'll need to add a separate paragraph to explain the other option ("Download selected"), to which the provider cli instructions don't apply

Author: Good point. This will be added when I can test out the feature.

You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
:::

> ### Download multiple files

Reviewer 2: ```suggestion
## Download multiple files
```

You can download multiple files via the user interface or using a script in your cloud provider's CLI. 

> :::note

Reviewer 2: I don't think this is correct. Downloading multiple folders/files works, but the folder structure is not preserved in non-Chromium based browsers.

Reviewer 3: @robnewman 

> FF/Safari download of folders doesn't preserve the folder-file structure but instead concatenates the folder hierarchy and file name. This is an acceptable work around for now.

What does the concatenated filename look like? Thanks!

> You can download multiple files, up to a maximum of 1,000 files, from the user interface or using a script in your cloud provider's CLI.

Reviewer: this is not correct, the 1000 files limit doesn't apply to the "download with script" flow, only to the download through browser / UI flow

Reviewer 3: I've updated this, thanks!

The browser download functionality currently does not work in non-Chromium browsers.
:::

> Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**. 

Reviewer 2: See [Slack discussion](https://seqera.slack.com/archives/C06B6RRGVCZ/p1714411320345289) on removing the `Actions` column

> Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**. 

Reviewer 2: ```suggestion
Open the cloud bucket that you'd like to download files and folder from. By default, download the contents of the current directory by choosing **Download current directory**. Alternatively, use checkboxes to select specific files and folders, and choose the **Download** button: you can **Download files** via the browser or **Download using code**. 
```

The code snippet provided is specific to the cloud provider you've configured. You may be prompted to authenticate during the download process. Refer to your cloud provider's documentation for troubleshooting credential-related issues:

- [GCP](https://cloud.google.com/sdk/gcloud/reference/storage)
- [AWS](https://docs.aws.amazon.com/cli/latest/reference/s3/)
- [Azure](https://learn.microsoft.com/en-us/cli/azure/storage?view=azure-cli-latest)

> ### AWS S3 CORS configuration

Reviewer 2: There needs to be a CORS policy applied for multiple file downloads too. See the validation [tests](https://github.com/seqeralabs/tower-backlog/issues/1270).

Reviewer 3: @robnewman, is this a request to incorporate those steps here? Or just to note that a user needs to deal with CORS policies somehow? Thanks!

Reviewer 2: It would be helpful to have examples for sure.

You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
:::

## Download multiple files

> You can download up to a maximum of 1,000 files using the browser interface, or an unlimited number of files with the auto-generated download script that uses your cloud provider's CLI and credentials.

Reviewer: small note: this `1000` limit is configurable. Its currently set to that value in theory it would be possible to change it. Not sure if these kind of details are usually included in the docs

Author: We can certainly add a note indicating that this is a configurable limit but then we'd need to have some sort of maximum value. It's not ideal to say it's adjustable without knowing what the upper limit is, imo.

I propose we let support know about this in case customers want to change the value. @alvaromartmart, where is this value found so that it can be changed?

Author: This can be a separate PR/discussion btw. 

Tagging @starnsd for visibility into this thread.

Author: Created https://seqera.atlassian.net/browse/EDU-161 to track this and resolving this conversation so we can merge.

You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
:::

### Download multiple files using a script

> If you are a Data Studios private preview customer, you can download multiple files either via the user interface or with a script.

Reviewer: this feature is not related to Data Studios
```suggestion
You can download multiple files either via the user interface or with a script.
```
